### PR TITLE
#45 Add typing capabilities to date/time inputs

### LINKED ISSUE #45: Possibility of manual entry in date/time inputs
## Basic information

* Tiller version: 1.2.0
* Module: date

## Enhancement description

- editing through the input field
- the possibility of an input mask
- date/time picking on one popover instead of tab (remove analogue clock)

### Current behaviour

- manual editing within the field is not possible
- input mask does not exist on this input field
- the date and time are selected through separate tabs

### PR DESCRIPTION
**Pull request by fbeceic**
## Basic information

* Tiller version: 1.2.0
* Module: date

## Description

Add typing capabilities to following components:

- Date Input (Field)
- Date Range Input (Field)
- Time Input (Field)
- Date Time Input (Field)

### Summary

- [x] Add date/time utilities
- [x] Implement typing capabilities for Date Input (Field)
- [x] Implement typing capabilities for Date Range Input (Field)
- [x] Implement typing capabilities for Date Time Input (Field)
- [x] Implement typing capabilities for Time Input (Field)
- [x] Enhance code display for components
- [x] Update docs 

### Related issue

Closes #45

## Types of changes

- Enhancement (non-breaking change which enhances existing functionality)
- New feature (non-breaking change which adds functionality)
- Docs improvement

## Checklist

- [x] I have read the project's **CONTRIBUTING** document
- [x] I have checked my code with the project's static analysis tooling
- [x] I have formatted my code with the project's Prettier code-style configuration
- [x] I have checked my code for misspellings
- [x] I have organized my changes in easy-to-follow commits
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added a story to cover my changes
- [x] All new and existing story tests pass

### CODE REVIEW

File: libs/date/src/DateInput.tsx [hunk 1]
```diff
@@ -230,14 +225,14 @@ export default function DateInput({
   };
 
   const onChange = (value: string) => {
-    const dateValue = formatDate(value, lang);
+    const dateValue = formatDate(value, formattedDateFormat || getDateFormatByLang(lang));
```
Comment (jtomic-croz) [inline libs/date/src/DateInput.tsx:228]: move this to variable instead of writing it 3 times :)

File: libs/date/src/DateInput.tsx [hunk 2]
```diff
@@ -311,24 +308,29 @@ function DateInputInput({
...
+    if (props.placeholder !== undefined) {
+      return props.placeholder;
+    }
+    if (dateFormat) {
+      return dateFormat;
+    }
+    if (showMaskOnEmpty) {
+      return undefined;
+    }
+    return lang === "en" ? "mm/dd/yyyy" : "dd. mm. yyyy.";
```
Comment (jtomic-croz) [inline libs/date/src/DateInput.tsx:321]: should't you use getDateFormatByLang here as well?

File: libs/date/src/utils.tsx [hunk 3]
```diff
@@ -258,23 +200,74 @@ export const timeMask = (value: string | null, twelveHours?: boolean) => {
...
+  const monthPosition = format.indexOf("MM");
+  const dayPosition = format.indexOf("dd");
+  const hourPosition = format.indexOf("HH");
+  const minutePosition = format.indexOf("mm");
+
+  const generatedMask: (string | RegExp)[] = [];
+  format.split("").forEach((char, index) => {
+    if (char === "M") {
+      if (monthPosition !== index) {
+        generatedMask.push(determineMonthInput("en", value, index - 1));
```
Comment (jtomic-croz) [inline libs/date/src/utils.tsx:213]: Why this depends on language?
second M is always depends on the value of first M, regardless of the language.

File: libs/date/src/DateInput.tsx [hunk 4]
```diff
@@ -318,7 +318,7 @@ function DateInputInput({
     if (showMaskOnEmpty) {
       return undefined;
     }
-    return lang === "en" ? "mm/dd/yyyy" : "dd. mm. yyyy.";
+    return getDateFormatByLang(lang).toLowerCase();
```
Comment (jtomic-croz) [inline libs/date/src/DateInput.tsx:321]: shouldn't this be `finalDateFormat.toLowerCase()`?

Comment (fbeceic) [reply libs/date/src/DateInput.tsx:321]: I don't have access to that variable anymore since it's defined in a parent component.

**Review by jtomic-croz — APPROVED** [review]

**Pull request merged**